qiskit/dagcircuit/dagcircuit.py

```diff
@@ -610,14 +610,20 @@ def compose(self, other, edge_map=None, qubits=None, clbits=None, front=False, i
             return None
 
     def idle_wires(self):
-        """Return idle wires.
+        """Return idle wires (wires which only contain barriers are idle).
 
         Yields:
             Bit: Bit in idle wire.
         """
         for wire in self._wires:
+            barrier_counter = 0
             nodes = self.nodes_on_wire(wire, only_ops=False)
-            if len(list(nodes)) == 2:
+            length = 0
+            for x in list(nodes):
+                length += 1
+                if x.name == 'barrier':
```

Comment:
In general, it's better when variable names a representative of their content. `node` is prefered over `x`.

Reply:
Ok clearer variables. I do have a tendency to assign those to random characters like x or i. Thank you for catching this 😂